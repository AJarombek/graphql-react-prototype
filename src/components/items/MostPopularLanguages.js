/**
 * MostPopularLanguages component which displays the most commonly used languages in repos.
 * @author Andrew Jarombek
 * @since 4/21/2020
 */

import React, { useEffect, useState } from 'react';
import { getTopLanguage } from '../../datasource/GraphQL';
import { getLanguagesSortedByOccurrence } from './githubUtils';

const MostPopularLanguages = () => {
  const [mostPopularLanguages, setMostPopularLanguages] = useState([]);
  const [error, setError] = useState(null);

  const generateTopLanguages = (repositories) => {
    const sortedLanguages = getLanguagesSortedByOccurrence(repositories);

    setMostPopularLanguages(sortedLanguages.slice(0, 5));
    setError(null);
  };

  useEffect(() => {
    async function getGraphQLResult() {
      const result = await getTopLanguage('AJarombek');

      if (result.data.data) {
        generateTopLanguages(result.data.data.user.repositories.edges);
      } else {
        setError(result.data.errors[0].message);
      }
    }

    getGraphQLResult();
  }, []);

  return (
    <div className="items most-popular-languages">
      {error ? (
        <div className="error">
          <h6>{error}</h6>
        </div>
      ) : (
        <>
          <h2>Most Popular Languages All-Time</h2>
          {mostPopularLanguages.map((language) => (
            <div className="language" key={language.name}>
              <p>{language.name}</p>
              <p>{language.occurrences}</p>
            </div>
          ))}
        </>
      )}
    </div>
  );
};

export default MostPopularLanguages;
