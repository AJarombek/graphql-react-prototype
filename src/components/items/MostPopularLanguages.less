/**
 * Stylesheet for the MostPopularLanguages component.
 * Author: Andrew Jarombek
 * Date: 4/21/2020
 */

@import "../../vars";

.most-popular-languages {
  grid-area: popLang;
}
