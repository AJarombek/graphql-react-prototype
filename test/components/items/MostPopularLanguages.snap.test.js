/**
 * Snapshot test for the MostPopularLanguages component.
 * @author Andrew Jarombek
 * @since 4/25/2020
 */

import React from 'react';
import MostPopularLanguages from '../../../src/components/items/MostPopularLanguages';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<MostPopularLanguages />).toJSON();
  expect(tree).toMatchSnapshot();
});
