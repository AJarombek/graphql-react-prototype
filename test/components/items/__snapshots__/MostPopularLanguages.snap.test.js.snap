// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items most-popular-languages"
>
  <h2>
    Most Popular Languages All-Time
  </h2>
</div>
`;
